tsdb/wal/wal.go

```diff
@@ -593,6 +605,7 @@ func (w *WAL) Log(recs ...[]byte) error {
 // - the record is bigger than the page size
 // - the current page is full.
 func (w *WAL) log(rec []byte, final bool) error {
+	defer w.metrics.writesTotal.Inc()
```

Comment:
We are measuring `writesFailed` in `Log`, so it will lead to confusion if anyone is trying to relate `writesFailed` with `writesTotal`. (Additionally, we should not have defer `defer` else it is possible to have `writesCompressed` > `writesTotal` which doesn't make sense :) )
```suggestion
	w.metrics.writesTotal.Inc()
```

Reply:
Yes, after review, the defer doesn't fit.